javascript/code-challenges/challenges-02.test.js

```diff
@@ -0,0 +1,322 @@
+
+CHALLENGE 1 - Review
+
+Write a function named raisedToTheThird that takes in an array of numbers and returns a new array of each of those numbers raised to the 3rd power (hint: look up Math.pow()). Use forEach to solve this problem.
+
+------------------------------------------------------------------------------------------------ */
+
+const raisedToTheThird = (arr) => {
+  // Solution code here...
+  const newArr = [];
+  arr.forEach(num => newArr.push(Math.pow(num,3)));
+  return newArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named addOne that, given an array of numbers, uses map to return a new array with each value simply incremented by 1. 
+------------------------------------------------------------------------------------------------ */
+
+const addOne = (arr) => {
+  // Solution code here...
+  return arr.map(num => num +1);
+
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named addQuestion that, given an array of strings, uses map to return a new array containing each string followed by a question mark character.
+------------------------------------------------------------------------------------------------ */
+
+const addQuestion = (arr) => {
+  // Solution code here...
+  return arr.map(str => str + '?');
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named forLoopTwoToThe that, given an array of integers as input, iterates over the array and returns a new array. The returned array should contain the result of raising 2 to the power of the original input element.
+
+You may choose to complete this challenge using a for loop, for...in syntax, or for...of syntax.
+
+For example, twoToThe([1,2,3]) returns [2,4,8] because 2 ^ 1 = 2, 2 ^ 2 = 4, and 2 ^ 3 = 8.
+------------------------------------------------------------------------------------------------ */
+
+const forLoopTwoToThe = (arr) => {
+  // Solution code here...
+  // return arr.map(int => Math.pow(2,int));
+  const newArr = [];
+
+  for (let i = 0; i < arr.length; i++){
+    const int = arr[i];
+    newArr.push(Math.pow(2,int));
+  }
+  return newArr;
+};
+
+
+
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named forEachTwoToThe that produces the same output as your forLoopTwoToThe function from challenge 4, but uses forEach instead of a for loop.
+------------------------------------------------------------------------------------------------ */
+
+const forEachTwoToThe = (arr) => {
+  // Solution code here...
+  const newArr = [];
+
+  arr.forEach(int => newArr.push(Math.pow(2,int)));
+  return newArr;
+};
+
```

Comment:
I know it says to use Math.pow, but ES6 also added the exponent operator "**" which I prefer.
```js
newArr.push(2 ** int);
```

Reply:
This is amazing!! Thanks!